src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -155,6 +155,19 @@ private async Task StartHostAsync(CancellationToken cancellationToken, int attem
             finally
             {
                 _hostStartSemaphore.Release();
+                await RestartIfHostOfflineChanged();
```

Comment:
Why doing this in both places? Can't we just do the check once?

Reply:
Good to safeguard even as part of the start, no?

Like we do it during a restart and a fresh start.